src/features/pageObjects/reducers/addPageObject.thunk.js

```diff
@@ -10,13 +10,33 @@ import {
 import { defaultLibrary } from "../../locators/types/generationClasses.types";
 import { getPageAttributes, isPONameUnique } from "../utils/pageObject";
 import { getClassName } from "../utils/pageObjectTemplate";
+import { LocatorType } from "../../../common/types/common";
+import { LocalStorageKey } from "../../../common/utils/const";
 
-export const addPageObj = createAsyncThunk("pageObject/addPageObj", async (payload, thunkAPI) => {
+export const addPageObj = createAsyncThunk("pageObject/addPageObj", async (payload, { getState }) => {
   const res = await getPageAttributes();
   const { title, url } = res[0].result;
   const className = getClassName(title);
-  const lastSelectedLibrary = selectLastElementLibrary(thunkAPI.getState());
-  const lastSelectedLocatorType = selectLastLocatorType(thunkAPI.getState());
+
+  const state = getState();
+
+  let lastSelectedLibrary;
+  let lastSelectedLocatorType;
+
+  if (localStorage.getItem(LocalStorageKey.Library)) {
+    lastSelectedLibrary = localStorage.getItem(LocalStorageKey.Library);
+  } else {
+    lastSelectedLibrary = selectLastElementLibrary(state) || defaultLibrary;
+    localStorage.setItem(LocalStorageKey.Library, lastSelectedLibrary);
```

Comment:
why do you save the value at creating PO, I'm sure we need to save it only when user change it